src/Filter/Uglifyjs.php

```diff
@@ -50,8 +64,33 @@ class Uglifyjs extends AssetFilter
     public function output($filename, $input)
     {
         $cmdSep = $this->_settings['version'] <= 1 ? ' - ' : '';
-        $cmd = $this->_settings['node'] . ' ' . $this->_settings['uglify'] . $cmdSep . $this->_settings['options'];
         $env = array('NODE_PATH' => $this->_settings['node_path']);
+        $cmd =
+            $this->_settings['node'] . ' ' .
+            $this->_settings['uglify'] . $cmdSep .
+            $this->_settings['options'];
+
+        if ($this->_settings['create_map']) {
+            $files = implode(' ', $this->files);
+            $path_parts = pathinfo($filename);
+            $this->_settings['options'] = str_replace(
+                '{{FILENAME}}',
+                $path_parts['filename'],
+                $this->_settings['options']
+            );
+            $this->_settings['source_map'] = str_replace(
+                '{{FILENAME}}',
+                $path_parts['filename'],
+                $this->_settings['source_map']
+            );
+            $cmd =
+                $this->_settings['node'] . ' ' .
+                $this->_settings['uglify'] . ' ' .
+                $files . ' ' .
+                $this->_settings['options'];
+            $cmd .= ' ' . $this->_settings['source_map'];
+            $input = '';
```

Comment:
Would you be able to add a test case covering the usage of these options? Right now its not clear on how they would be used as there is no documentation or test cases.

Reply:
I will update this in the next days to cover the UglifyJS options better and then write some tests.

Reply:
Awesome. I noticed the comment you posted about how this could be used, and I would prefer if we had a higher level API for those options. Perhaps something like

```ini
[filter_Uglifyjs]
create_source_map = true
source_map_output = WEBROOT/cache_js/{{FILENAME}}.js.map
source_map_base_url = http://boom.dev/cache_js/
```

Based on this information we can generate the actual options that uglify needs.